modules/select2_facets/src/Plugin/facets/widget/Select2Widget.php

```diff
@@ -37,26 +35,14 @@ class Select2Widget extends WidgetPluginBase implements ContainerFactoryPluginIn
    */
   protected $keyValueStore;
 
-  /**
-   * {@inheritdoc}
-   */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, Request $request, KeyValueStoreInterface $key_value_store) {
-    parent::__construct($configuration, $plugin_id, $plugin_definition);
-    $this->request = $request;
-    $this->keyValueStore = $key_value_store;
-  }
-
   /**
    * {@inheritdoc}
    */
   public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
-    return new static(
-      $configuration,
-      $plugin_id,
-      $plugin_definition,
-      $container->get('request_stack')->getCurrentRequest(),
-      $container->get('keyvalue')->get('entity_autocomplete')
-    );
+    $plugin = new static($configuration, $plugin_id, $plugin_definition);
```

Comment:
I know this is kinda useless because when the parent constructor changes, we have to change this call here, too.

The reason is that PluginBase does not implement ContainerFactoryPluginInterface.